src/seml/database.py

```diff
@@ -19,7 +23,46 @@ def get_collection(collection_name, mongodb_config=None, suffix=None):
     return db[collection_name]
 
 
-def get_mongo_client(db_name, host, port, username, password, **kwargs):
+def retried_and_locked_ssh_port_forward(
+        retries_max=6, retries_delay=1, lock_file='~/seml_ssh.lock', lock_timeout=30, **ssh_config):
```

Comment:
Can we move these defaults to `settings.py`. The lock_file's default should be somewhere in SETTINGS.TMP_DIRECTORY` since the home directory is mostly distributed via nfs, we don't want to lock across nodes.